components/dashboard/src/data/ide-options/ide-options-query.ts

```diff
@@ -76,20 +76,26 @@ export const useAllowedWorkspaceEditorsMemo = (configurationId: string | undefin
         // So we will filter it out
         isLoading = isLoadingInstallationCls || isLoadingConfiguration;
     }
+    const depItem = {
+        t1: installationOptions,
+        t2: options?.ignoreScope,
+        t3: orgSettings,
+        t4: configuration?.workspaceSettings?.restrictedEditorNames,
+    };
     const data = useMemo(() => {
         return getAllowedWorkspaceEditors(
             installationOptions,
             orgSettings,
             configuration?.workspaceSettings?.restrictedEditorNames,
             options,
         );
+
+        // react useMemo is using `Object.is` to compare dependencies so array / object will make re-render re-call useMemo,
+        // see also https://react.dev/reference/react/useMemo#every-time-my-component-renders-the-calculation-in-usememo-re-runs
+        // we only use basic types like string here
+        //
         // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [
-        installationOptions,
-        options?.ignoreScope,
-        orgSettings,
-        configuration?.workspaceSettings?.restrictedEditorNames,
-    ]);
+    }, [JSON.stringify(depItem)]);
```

Comment:
Better to use this https://github.com/kentcdodds/use-deep-compare-effect, `JSON.stringify` will change if order inside properties change too

Reply:
@jeanp413  it still consider value changed, `JSON.stringify` seems to be more constant for our case
<img width="1579" alt="image" src="https://github.com/gitpod-io/gitpod/assets/20944364/eedcdde0-eb8e-44b0-8c31-c5eb8cc072dc">

Reply:
Just tested it in https://github.com/gitpod-io/gitpod/tree/jp/intense-tuna it works